src/App.js

```diff
@@ -0,0 +1,56 @@
+import {
+  BrowserRouter as Router,
+  Switch,
+  Route,
+} from 'react-router-dom';
+import Header from './components/Nav';
+import Book from './components/Books';
+import NewBook from './components/NewBook';
+import Categories from './components/Categories';
+
+const books = [
+  {
+    id: 1,
+    category: 'Action',
+    name: 'The Hunger Games',
+    author: 'Suzanne Collins',
+    percentage: '64%',
+    chapter: 'Chapter 17',
+  },
+  {
+    id: 2,
+    category: 'Science Fiction',
+    name: 'Dune',
+    author: 'Frank Herbert',
+    percentage: '8%',
+    chapter: 'Chapter 3: "A Lesson Learned"',
+  },
+  {
+    id: 3,
+    category: 'Economy',
+    name: 'Capital in the Twenty-First Century',
+    author: 'Suzanne Collins',
+    percentage: '0%',
+    chapter: 'Introduction',
+  },
+];
+
+const App = () => {
+  const bookList = books.map((books) => <Book key={books.id} info={books} />);
+  return (
+    <Router>
+      <Header />
+      <Switch>
+        <Route exact path="/">
+          { bookList }
+          <NewBook />
```

Comment:
Let's try to maintain the SOLID principle and refactor this `App.js` component a little bit.
- [ ] Firstly, there should be a `BookList` component that renders the books. Here you are rendering the `Book` component directly onto the App component, but this should be done in the `BookList` component which you are yet to create. Kindly create the `BookList` component and move this `books` array and the mapping into the new `BookList` component.
- [ ] The `BookList` component should also render the **`NewBook`** component. So also extract line 46 into `BookList`.
- [ ] Now render the `BookList` in `App` as the root path in place of line 45 and 46:
  
  ```
  { bookList }
  <NewBook />
  ```
  Your App will work just fine and render as you currently have it.